manager/orchestrator/updater.go

```diff
@@ -104,7 +104,7 @@ func (u *Updater) Run(ctx context.Context, service *api.Service, tasks []*api.Ta
 	dirtyTasks := []*api.Task{}
 	for _, t := range tasks {
 		if !reflect.DeepEqual(service.Spec.Task, t.Spec) ||
-			!reflect.DeepEqual(service.Endpoint, t.Endpoint) {
+			!reflect.DeepEqual(service.Spec.Endpoint, t.Endpoint.Spec) {
```

Comment:
Hmm.. I also tested this whole thing in `dind` but not sure why t.Endpoint can be `nil`. we do initialize it in orchetrator